compiler-rt/test/nsan/nan.cpp

```diff
@@ -0,0 +1,19 @@
+// RUN: %clangxx_nsan -O0 -g %s -o %t
```

Comment:
need to make this option work with `NSAN_OPTIONS=halt_on_error=0` and `NSAN_OPTIONS=halt_on_error=1`
(the behavior should be similar to the current consistency check)
(and cover by tests)